tests/test_depsolver.py

```diff
@@ -287,6 +289,14 @@ def test_run(pulp):
                 ]
             )
 
+            # INTERMITTENT BUG, OUTPUT SHOULD NOT BE DUPLICATED
+            babels = [item for item in depsolver.output_set if item.name == 'babel']
+            if len(babels) == 2:
+                # if entered, the below assertion will fail (true duplicate)
+                assert babels[0]._unit != babels[1]._unit
+                # when UbiUnits are compared (in sets) the objects are compaired, not their content
+                # so we must look at the underlying unit to truly get unique items in output_set
+
```

Comment:
@rbikar I've been struggling with an issue I've just confirmed to be a bug. That is, assuming we /don't/ want duplicate items in output_set.

I can try to implement a fix here but it may be more appropriate for a separate commit.

Reply:
Something like this works;
```
    out_dict = {}
    for item in self.output_set:
        out_dict.update({item.filename or item.name: item})
    self.output_set = set(out_dict.values())
```
Not sure how robust it really needs to be though

Reply:
I guess fixing this _TODO_ https://github.com/release-engineering/ubi-manifest/blob/87f9ec909cd34e308dc1bdd1435437c7d0e3d3b8/ubi_manifest/worker/tasks/depsolver/models.py#L26  should fix the duplicates problems. I think everything coming from pulplib should be now hashable, there was a problem with modulemd or modulemd_defaults.

Reply:
Ah, yeah, seems like that would work. I'll give it a try.

Reply:
@rbikar, that's worked to dedup though I had to add an __eq__ as well